controllers/kuadrant_controller.go

```diff
@@ -320,23 +322,32 @@ func (r *KuadrantReconciler) getIstioConfigObjects(ctx context.Context, logger l
 	var configsToUpdate []common.ConfigWrapper
 
 	iop := &iopv1alpha1.IstioOperator{}
-	iopKey := client.ObjectKey{Name: controlPlaneProviderName(), Namespace: controlPlaneProviderNamespace()}
-	if err := r.GetResource(ctx, iopKey, iop); err != nil {
-		logger.V(1).Info("failed to get istiooperator object", "key", iopKey, "err", err)
-		if apimeta.IsNoMatchError(err) {
-			// return nil and nil if there's no istiooperator CRD, means istio is not installed
-			return nil, nil
-		} else if !apierrors.IsNotFound(err) {
-			// return nil and err if there's an error other than not found (no istiooperator CR)
-			return nil, err
-		}
-	} else {
+	istKey := client.ObjectKey{Name: controlPlaneProviderName(), Namespace: controlPlaneProviderNamespace()}
+	err := r.GetResource(ctx, istKey, iop)
+	if err == nil || apierrors.IsNotFound(err) {
 		configsToUpdate = append(configsToUpdate, istio.NewOperatorWrapper(iop))
+	} else if !apimeta.IsNoMatchError(err) {
+		logger.V(1).Info("failed to get istiooperator object", "key", istKey, "err", err)
+		return nil, err
+	} else {
+		// Error is NoMatchError so check for Istio CR instead
+		ist := &istiov1alpha1.Istio{}
```

Comment:
Is this CR watched by the istio operator too?

Reply:
Yep it is